src/transport.js

```diff
@@ -82,10 +82,18 @@ module.exports = function (swarm) {
       const createListeners = multiaddrs.map((ma) => {
         return (cb) => {
           const listener = transport.createListener(handler)
-
-          listener.once('error', cb)
+          let err = false
+          listener.once('error', (err) => {
+            err = true
+            cb(err)
+          })
 
           listener.listen(ma, () => {
+            // Already errored out
+            if (err) {
+              return
+            }
+
```

Comment:
We can't, that's what it was before, which doesn't catch the case where an error happens but the listen callback still happens